sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -474,14 +475,24 @@ void ng_ipv6_netif_init_by_dev(void)
                            sizeof(if_type)) != -ENOTSUP) &&
             (if_type == NG_NETTYPE_SIXLOWPAN)) {
             uint16_t src_len = 8;
-            DEBUG("Set 6LoWPAN flag\n");
+            uint16_t max_frag_size = UINT16_MAX;
+
+            DEBUG("ipv6 netif: Set 6LoWPAN flag\n");
             ipv6_ifs[i].flags |= NG_IPV6_NETIF_FLAGS_SIXLOWPAN;
             /* use EUI-64 (8-byte address) for IID generation and for sending
              * packets */
             ng_netapi_set(ifs[i], NETCONF_OPT_SRC_LEN, 0, &src_len,
                           sizeof(src_len)); /* don't care for result */
-        }
 
+            if (ng_netapi_get(ifs[i], NETCONF_OPT_MAX_PACKET_SIZE,
+                              0, &max_frag_size, sizeof(max_frag_size)) < 0) {
+                /* if error we assume it works */
```

Comment:
I just think that it is more likely for a link layer to being able to carry 1--2kBytes instead of being able to carry 65kBytes payload.

And yes, I lost track of what this PR is essentially doing. I cannot do a proper review (without reading and understanding the whole fragmentation code again). So, I would say, if no one else is willing to review, let's trust the outcome, which seems to be good according to @PeterKietzmann.